aea/registries/base.py

```diff
@@ -608,13 +609,14 @@ def handle_internal_messages(self) -> None:
         :return: None
         """
         while not self.decision_maker_out_queue.empty():
-            tx_message = self.decision_maker_out_queue.get_nowait()  # type: Optional[TransactionMessage]
+            tx_message = cast(TransactionMessage, self.decision_maker_out_queue.get_nowait())
+            tx_message.counterparty = cast(Address, DECISION_MAKER)
```

Comment:
this should be moved below the next if statement

Reply:
In fact, we don;'t need to set it at all as it is clear that transaction messages come from the same agent.